### DISCUSSION TITLE
How to have config changes take effect?

### DISCUSSION THREAD

**Discussion by RedBeard0531** [neoclide/coc.nvim#4171] [Q&A] [answered]
When using `:CocConfig` to edit the config, I notice that sometimes the modifications take effect as soon as I save that file, but sometimes I need to run `:CocRestart`. I haven't found a perfect pattern to this. It seems more likely to work the first time I edit and save the file, than later times, but it doesn't always work on the first time. Is there some reason for this? Is there a lighter-weight way to reload config without needing to restart coc and especially all of the LSP servers?

**Comment by chemzqm** [comment]
Most configurations should work after save the config file, but some would require `:CocRestart` to take effect

**Reply by RedBeard0531** [reply]
Then maybe something is broken for me. For example, changing the values of `"suggest.completionItemKindLabels"` sometimes requires a restart and sometimes takes effect on save. I have not found any rhyme or reason to when it does or doesn't work.

How is coc detecting the config file change? I wonder if it is malfunctioning on my system (linux, neovim). Or perhaps something is preventing an autocmd from being delivered?

**Reply by chemzqm** [reply]
I guess not, there're multiple places using `suggest.completionItemKindLabels`, so the check for change could be missed.

**Comment by chemzqm** [comment, marked as answer]
Open configruation doc by `:h coc-config` and search for `restart`, consider send bug report if the configuration requires restart coc.nvim if it should not.
